src/EnergyPlus/AirTerminalUnit.hh

```diff
@@ -0,0 +1,118 @@
+#ifndef AirTerminalUnit_hh_INCLUDED
+#define AirTerminalUnit_hh_INCLUDED
+
+// C++ Headers
+#include <string>
+
+// EnergyPlus Headers
+#include <EnergyPlus.hh>
+
+namespace EnergyPlus {
+
+// types of air terminal units, refactored from old DataDefineEquip
+enum AirTerminalUnitType {
+	notYetDetermined,
+	dualDuctConstVolume,
+	dualDuctVAV,
+	singleDuctVAVReheat,
+	singleDuctConstVolReheat,
+	singleDuctVAVNoReheat,
+	singleDuct_SeriesPIU_Reheat,
+	singleDuct_ParallelPIU_Reheat,
+	singleDuct_ConstVol_4PipeInduc,
+	singleDuctVAVReheatVSFan,
+	singleDuctCBVAVReheat,
+	singleDuctCBVAVNoReheat,
+	singleDuctConstVolCooledBeam,
+	dualDuctVAVOutdoorAir,
+	singleDuctUserDefined,
+	singleDuctInletATMixer,
+	singleDuctSupplyATMixer,
+	singleDuctConstVolFourPipeBeam
+};
+
+// base class for all air distribution units.  zone air terminals for connecting to central air handlers
+class AirTerminalUnit
+{
+
+protected: // Creation
+
+	// Default Constructor
+	AirTerminalUnit() :
+		terminalType( notYetDetermined ),
+		aDUNum( 0 ),
+		airAvailSchedNum( 0 ),
+		airAvailable( false ),
+		vDotDesignPrimAir( 0.0 ),
+		vDotDesignPrimAirWasAutosized( false ),
+		mDotDesignPrimAir( 0.0 ),
+		airInNodeNum( 0 ),
+		airOutNodeNum( 0 ),
+		zoneIndex( 0 ),
+		zoneNodeIndex( 0 )
+	{}
+
+	// Copy Constructor
+	AirTerminalUnit( AirTerminalUnit const & ) = default;
+
+	// Move Constructor
+#if !defined(_MSC_VER) || defined(__INTEL_COMPILER) || (_MSC_VER>=1900)
+	AirTerminalUnit( AirTerminalUnit && ) = default;
+#endif
+
+public: // Creation
+
+	// Destructor
+	virtual
+	~AirTerminalUnit()
+	{}
+
+protected: // Assignment
+
+	// Copy Assignment
+
+	AirTerminalUnit &
+	operator =( AirTerminalUnit const & ) = default;
+
+	// Move Assignment
+#if !defined(_MSC_VER) || defined(__INTEL_COMPILER) || (_MSC_VER>=1900)
+	AirTerminalUnit &
+	operator =( AirTerminalUnit && ) = default;
+#endif
+
+public: // Methods		REMOVE ANY OF THESE THAT AREN'T COMMON (WITH SAME ARGS) TO ALL SUB-TYPES
+
+	// for unit tests
+	virtual
+	void
+	clear_state() = 0;
```

Comment:
@EnergyArchmage I don't think there should be a clear_state() within classes. The clear_state() was a (hacky) way to fix unit testing with so many namespace variables. You won't need to clear any state within an object, and if you do, it should be in the destructor (should be rare we need to do this).